app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -144,10 +143,35 @@ public void afterTextChanged(Editable s) {
         }
     }
 
+    /**
+     * Uses previously saved theme if there is any, if not then uses default.
+     */
+    private void prepareTheme() {
+
+        String currentTheme = getCurrentTheme();
+
+        themeListPreference.setSummary(getThemeSummary(currentTheme));
+        themeListPreference.setValue(currentTheme);
```

Comment:
You can get rid of setValue, this should be the last comment!

Reply:
@macgills I kept it because it is important for the first time when the user installs the app!

Reply:
`defaultValue` in the xml should take care of it, maybe you didn't have that set when you first added this line

Reply:
Since I removed the default value ("0") from the strings.xml
I think it is better to keep it in java code, in case we decide to change it to something else later we might forget to change it in XML but this is just my thinking.

Reply:
This is why I hate PreferenceFragment. I'd just keep default value in the xml.

The only other way to work around is a subclass of Preference or black magic with gradle

Reply:
Done!